xds/internal/balancer/clusterresolver/priority_test.go

```diff
@@ -1,834 +0,0 @@
-	"google.golang.org/grpc/xds/internal/balancer/priority"
-	"google.golang.org/grpc/xds/internal/balancer/wrrlocality"
-	xdstestutils "google.golang.org/grpc/xds/internal/testutils"
-	"google.golang.org/grpc/xds/internal/testutils/fakeclient"
-	"google.golang.org/grpc/xds/internal/xdsclient"
-)
-
-var (
-	testClusterNames    = []string{"test-cluster-1", "test-cluster-2"}
-	testSubZones        = []string{"I", "II", "III", "IV"}
-	testEndpointAddrs   []string
-	wrrLocalityLBConfig = &internalserviceconfig.BalancerConfig{
-		Name: wrrlocality.Name,
-		Config: &wrrlocality.LBConfig{
-			ChildPolicy: &internalserviceconfig.BalancerConfig{
-				Name: "round_robin",
-			},
-		},
-	}
-)
-
-const testBackendAddrsCount = 12
-
-func init() {
-	for i := 0; i < testBackendAddrsCount; i++ {
-		testEndpointAddrs = append(testEndpointAddrs, fmt.Sprintf("%d.%d.%d.%d:%d", i, i, i, i, i))
-	}
-	balancergroup.DefaultSubBalancerCloseTimeout = time.Millisecond
-	clusterimpl.NewRandomWRR = testutils.NewTestWRR
-	weightedtarget.NewRandomWRR = testutils.NewTestWRR
-	balancergroup.DefaultSubBalancerCloseTimeout = time.Millisecond * 100
-}
-
-func setupTestEDS(t *testing.T, initChild *internalserviceconfig.BalancerConfig) (balancer.Balancer, *testutils.TestClientConn, *fakeclient.Client, func()) {
-	xdsC := fakeclient.NewClientWithName(testBalancerNameFooBar)
-	cc := testutils.NewTestClientConn(t)
-	builder := balancer.Get(Name)
-	edsb := builder.Build(cc, balancer.BuildOptions{Target: resolver.Target{URL: *testutils.MustParseURL("dns:///" + testEDSService)}})
-	if edsb == nil {
-		t.Fatalf("builder.Build(%s) failed and returned nil", Name)
-	}
-	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
-	defer cancel()
-	if err := edsb.UpdateClientConnState(balancer.ClientConnState{
-		ResolverState: xdsclient.SetClient(resolver.State{}, xdsC),
-		BalancerConfig: &LBConfig{
-			DiscoveryMechanisms: []DiscoveryMechanism{{
-				Cluster: testClusterName,
-				Type:    DiscoveryMechanismTypeEDS,
-			}},
-			xdsLBPolicy: *wrrLocalityLBConfig,
-		},
-	}); err != nil {
-		edsb.Close()
-		t.Fatal(err)
-	}
-	if _, err := xdsC.WaitForWatchEDS(ctx); err != nil {
-		edsb.Close()
-		t.Fatalf("xdsClient.WatchEndpoints failed with error: %v", err)
-	}
-	return edsb, cc, xdsC, func() {
-		edsb.Close()
-	}
-}
-
-// When a high priority is ready, adding/removing lower locality doesn't cause
-// changes.
-//
-// Init 0 and 1; 0 is up, use 0; add 2, use 0; remove 2, use 0.
-func (s) TestEDSPriority_HighPriorityReady(t *testing.T) {
-	edsb, cc, xdsC, cleanup := setupTestEDS(t, nil)
-	defer cleanup()
-
-	// Two localities, with priorities [0, 1], each with one backend.
-	clab1 := xdstestutils.NewClusterLoadAssignmentBuilder(testClusterNames[0], nil)
-	clab1.AddLocality(testSubZones[0], 1, 0, testEndpointAddrs[:1], nil)
-	clab1.AddLocality(testSubZones[1], 1, 1, testEndpointAddrs[1:2], nil)
-	xdsC.InvokeWatchEDSCallback("", parseEDSRespProtoForTesting(clab1.Build()), nil)
```

Comment:
Interesting. This test does it through client and lets this balancer build out priority configuration from EDS. Priority_Test just inlines an UpdateClientConnState() with what this balancer would've built out. Yeah, makes more sense to have these tests over there.

Reply:
Yeah, we have tests in this package which verify that we build the appropriate priority LB configuration based on received updates. These are unit tests for now, could be changed to an e2e style test with the update coming through the xdsClient, but what we have is good enough for now I think.